### ISSUE TITLE
[AI Team] SearchBench Gold — relevance, requirement truth, and mutation gates

### ISSUE THREAD

**Issue by middlechildmzk** [middlechildmzk/sourcingos-unified#283]
Parent coordination: #282

## Owner shape
Benchmark/evaluation workstream. This workstream should avoid production parser/search implementation unless a benchmark harness seam explicitly requires it.

## Mission
Turn real recruiter incidents into permanent measurable release gates.

## Initial gold cases
1. **SQL Server DBA — Colorado Springs / Maximus**
   - preserve hard/technical/preferred truth
   - known candidates and internal routing
   - obvious false positives
   - geography truth
   - candidate-by-candidate QA/relevance labels
2. **Machine Learning Engineer — TechNova JD**
   - title/location/work-mode extraction
   - AND/OR alternative groups
   - no prose/salary/benefits contamination
   - no duplicated requirements
   - correct technical/preferred grouping

## Metrics
- requirement fidelity
- requirement-tier fidelity
- contamination rate
- unsupported-claim rate
- Precision@5 / @10 / @25
- nDCG@10
- known-qualified recall proxy
- duplicate-person rate
- geography-state accuracy
- unique useful/qualified canonical yield by source
- zero-result rate
- time to first useful candidate

## Mutation requirements
Every high-value guard should prove a negative mutation turns the benchmark red. Include at minimum:
- previous-plan contamination restored
- VALIDATE_EXISTING discovery authorization removed
- technical→hard promotion restored
- known location mismatch collapsed into unknown
- `TS/SCI` allowed to satisfy TypeScript / vice versa
- generic SQL allowed to satisfy Microsoft SQL Server
- discovery expansion allowed to satisfy a hard requirement
- name-only identity merge enabled

## Output
A deterministic SearchBench Gold harness suitable for CI and local development with zero paid API/provider calls.

**Comment by middlechildmzk** [comment]
### SearchBench Gold schema decision — separate intake truth from candidate relevance truth

Do not build one giant fixture format. The ML/JD parser benchmark and candidate-search benchmark test different contracts and should share IDs, not fields.

#### A. `RoleIntakeGoldV1`
Use after V57 lands:
```ts
type RoleIntakeGoldV1 = {
  id: string
  sourceText: string
  sourceDigest: string
  expectedFacts: Array<{ kind: string; value: string; sourceText: string }>
  expectedRequirements: Array<{
    id: string
    tier: 'hard' | 'technical' | 'preferred'
    sourceText: string
    logic?: { operator: 'AND' | 'OR'; members: string[] }
    constraints?: Array<{ kind: string; min?: number; value?: string }>
  }>
  expectedDisqualifiers: Array<{ sourceText: string }>
  expectedInstructions: Array<{ kind: 'search' | 'validation' | 'output' | 'known_candidate' | 'internal'; sourceText: string }>
  expectedNonRequirements: string[]
  allowedAmbiguousSpans: string[]
  forbiddenOutputs: string[]
}
```
Gold truth must quote literal source spans. Normalized values can be additional assertions, never replacements for grounding.

#### B. `CandidateRelevanceGoldV1`
Use for post-retrieval SearchBench:
```ts
type CandidateRelevanceGoldV1 = {
  benchmarkId: string
  candidateKey: string
  expectedClass: 'strong_lead' | 'worth_review' | 'weak_adjacent' | 'irrelevant' | 'known_duplicate' | 'internal'
  idealRankBand?: [number, number]
  requirementTruth: Record<string, 'MET' | 'PARTIAL' | 'UNKNOWN' | 'NOT_MET'>
  identityTruth?: 'distinct' | 'same_person_confirmed' | 'possible_same_person'
  geographyTruth?: 'match' | 'unknown' | 'mismatch' | 'relocation_explicit'
  notes: string
}
```
Candidate relevance labels must be judged from frozen evidence available in the fixture; no live provider lookup in CI.

#### C. Execution receipt fixture
Every search benchmark should also freeze expected routing semantics:
- discovery authorized yes/no
- selected source lanes
- expected zero-paid-call test mode
- raw discovered count
- same-source duplicates suppressed
- admitted count
- known/internal/suppressed count
- displayed count after filters

This prevents UI-count regressions from being confused with relevance regressions.

### Gemini torture suite handling
The 25 Gemini fixtures are **adversarial candidate fixtures, not automatic gold labels**. Import them only after reviewing expected truth. Keep especially:
- TS/SCI vs TypeScript
- SQL vs Microsoft SQL Server
- title/location separation
- Computer Science vs geography
- PyTorch OR TensorFlow
- AWS OR GCP OR Azure
- C++ vs C#
- prose-embedded explicit requirements
- boilerplate/EEO/benefits exclusion
- negative constraints
- internal sourcer instructions
- preferred→hard mutation
- Unicode/HTML ATS formatting
- multi-location parsing
- overlapping overall + skill-specific tenure
- non-solicit/employer exclusions
- literal-grounding/no invented skills

Do **not** freeze Gemini expectations that infer unstated onsite mode, clearance availability, education equivalence, years scope, or other recruiter truth not literally supported by the fixture.

### Metric denominator rules
- `Precision@K`: only default recruiter-visible external people slate; known/internal/suppressed audit lanes are reported separately.
- `requirement fidelity`: exact expected requirement IDs/logic/tier, not keyword overlap.
- `unsupported claim rate`: any displayed MET/SUPPORTED claim without admissible frozen evidence.
- `duplicate-person rate`: duplicate recruiter-visible provisional people, not raw provider observations.
- `unique qualified yield`: unique provisional canonicals in acceptable benchmark classes contributed by a source after same-source dedup; do not equate raw returned records with yield.
- `time to first useful candidate`: execution telemetry only in controlled integration/perf tests, not deterministic unit CI.

### Release-gate order
1. V57 Role Intake Gold #002 must pass before V57 release.
2. V58 requirement-state mutations (#291) must pass before evaluation-kernel release.
3. Candidate relevance/ranking metrics become blocking only once frozen candidate evidence slates exist for Gold #001/#002.
4. Provider-yield routing remains non-blocking until identity grouping and enough labeled searches exist.

This keeps SearchBench from becoming a benchmark that rewards the same heuristics it is supposed to catch.
